web-local/src/routes/(viz)/dashboard/[name]/+page.svelte

```diff
@@ -24,26 +25,45 @@
   $: ({ instanceId } = $runtime);
 
   $: filePaths = data.metricsView.meta?.filePaths as string[];
-  $: projectParserQuery = useProjectParser(queryClient, instanceId);
+  $: dashboard = useDashboard(instanceId, metricsViewName);
+  $: measures = $dashboard.data?.metricsView?.state?.validSpec?.measures ?? [];
+  $: projectParserQuery = useProjectParser(
+    queryClient,
+    instanceId,
+    measures.length > 0,
+  );
   $: dashboardFileHasParseError =
     $projectParserQuery.data?.projectParser?.state?.parseErrors?.filter(
       (error) => filePaths.includes(error.filePath as string),
     );
-
-  $: dashboard = useDashboard(instanceId, metricsViewName);
   $: mockUserHasNoAccess =
     $selectedMockUserStore && $dashboard.error?.response?.status === 404;
+
+  // Handle errors from dashboard YAML edits from an external IDE
+  $: if (dashboardFileHasParseError && dashboardFileHasParseError.length > 0) {
+    eventBus.emit("banner", {
+      type: "error",
+      message:
+        "Error parsing dashboard – you are viewing your last valid dashboard specification",
+      iconType: "alert",
+    });
+  }
+
+  $: if (mockUserHasNoAccess || measures.length === 0) {
+    // We want to close the banner before any of the ErrorPage is rendered
+    eventBus.emit("banner", null);
```

Comment:
We should also clear the banner when navigating away from the page. (Which is logic that probably should be built into the Banner Center at some point.)